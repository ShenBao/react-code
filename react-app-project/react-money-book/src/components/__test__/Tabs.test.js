import React from 'react'
import { shallow } from 'enzyme'
import { Tabs, Tab } from '../Tabs'

const props = {
  onTabChange: jest.fn(),
  activeIndex: 0,
}
let wrapper

describe('test Tabs and Tab Component', () => {
  beforeEach(() => {
    wrapper = shallow(
      <Tabs {...props}>
        <Tab>1</Tab>
        <Tab>2</Tab>
      </Tabs>
    )
  })
  it('should render the component to match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
  it('should render two Tab component, first one should be active', () => {
    expect(wrapper.find(Tab).length).toEqual(2)
    expect(wrapper.find('.nav-link').length).toEqual(2)
    expect(wrapper.state().activeIndex).toEqual(0)
    expect(wrapper.find('.nav-link').first().hasClass('active')).toEqual(true)
  })
  it('click the 2nd Tab should change the active stauts and trigger the right function', () => {
    wrapper.find('.nav-link').last().simulate('click', { preventDefault: () => {}})
    expect(wrapper.find('.nav-link').first().hasClass('active')).toEqual(false)
    expect(wrapper.find('.nav-link').last().hasClass('active')).toEqual(true)
    expect(wrapper.state().activeIndex).toEqual(1)
    expect(props.onTabChange).toHaveBeenCalledWith(1)
  })
})