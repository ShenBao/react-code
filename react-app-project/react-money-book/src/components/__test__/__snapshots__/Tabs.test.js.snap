// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Tabs and Tab Component should render the component to match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Tabs
    activeIndex={0}
    onTabChange={[Function]}
>
    <Unknown>
        1
    </Unknown>
    <Unknown>
        2
    </Unknown>
</Tabs>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <li
          className="nav-item"
>
          <a
                    className="nav-link active"
                    onClick={[Function]}
                    role="button"
          >
                    <Unknown>
                              1
                    </Unknown>
          </a>
</li>,
        <li
          className="nav-item"
>
          <a
                    className="nav-link"
                    onClick={[Function]}
                    role="button"
          >
                    <Unknown>
                              2
                    </Unknown>
          </a>
</li>,
      ],
      "className": "nav nav-tabs nav-fill my-4",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": ".0",
        "nodeType": "host",
        "props": Object {
          "children": <a
            className="nav-link active"
            onClick={[Function]}
            role="button"
>
            <Unknown>
                        1
            </Unknown>
</a>,
          "className": "nav-item",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Unknown>
              1
</Unknown>,
            "className": "nav-link active",
            "onClick": [Function],
            "role": "button",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "1",
            },
            "ref": null,
            "rendered": "1",
            "type": [Function],
          },
          "type": "a",
        },
        "type": "li",
      },
      Object {
        "instance": null,
        "key": ".1",
        "nodeType": "host",
        "props": Object {
          "children": <a
            className="nav-link"
            onClick={[Function]}
            role="button"
>
            <Unknown>
                        2
            </Unknown>
</a>,
          "className": "nav-item",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Unknown>
              2
</Unknown>,
            "className": "nav-link",
            "onClick": [Function],
            "role": "button",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "2",
            },
            "ref": null,
            "rendered": "2",
            "type": [Function],
          },
          "type": "a",
        },
        "type": "li",
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li
            className="nav-item"
>
            <a
                        className="nav-link active"
                        onClick={[Function]}
                        role="button"
            >
                        <Unknown>
                                    1
                        </Unknown>
            </a>
</li>,
          <li
            className="nav-item"
>
            <a
                        className="nav-link"
                        onClick={[Function]}
                        role="button"
            >
                        <Unknown>
                                    2
                        </Unknown>
            </a>
</li>,
        ],
        "className": "nav nav-tabs nav-fill my-4",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": ".0",
          "nodeType": "host",
          "props": Object {
            "children": <a
              className="nav-link active"
              onClick={[Function]}
              role="button"
>
              <Unknown>
                            1
              </Unknown>
</a>,
            "className": "nav-item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Unknown>
                1
</Unknown>,
              "className": "nav-link active",
              "onClick": [Function],
              "role": "button",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "1",
              },
              "ref": null,
              "rendered": "1",
              "type": [Function],
            },
            "type": "a",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": ".1",
          "nodeType": "host",
          "props": Object {
            "children": <a
              className="nav-link"
              onClick={[Function]}
              role="button"
>
              <Unknown>
                            2
              </Unknown>
</a>,
            "className": "nav-item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Unknown>
                2
</Unknown>,
              "className": "nav-link",
              "onClick": [Function],
              "role": "button",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "2",
              },
              "ref": null,
              "rendered": "2",
              "type": [Function],
            },
            "type": "a",
          },
          "type": "li",
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
